### ISSUE TITLE
operator invalid_date causes exception for date like "20220921"

### ISSUE THREAD

**Issue by JozefAerts** [cdisc-org/cdisc-rules-engine#126]
Rule SENDIG67 states that "ISO 8601 Extended format, which requires hyphen delimiters for the date components and colon delimiters for the time components, must be used for SDTM and SENDIG."

When I use a date that violates the rule, like "20220921" using the operator "invalid_date", an exception with the message:
"int() argument must be a string, a bytes-like object or a number, not 'NoneType'"

occurs.

**Comment by nhaydel** [comment]
This is likely an issue with the invalid_date operator defined here: https://github.com/cdisc-org/business-rules

**Comment by nhaydel** [comment]
@JozefAerts the issue for invalid_dates is resolved. The exception you are seeing is caused by the SEQ value being populated for some records and not for others. If you populate the SEQ column in your datasets it works. I have updated the invalid_date operator to better handle missing dates. I will create another issue for the SEQ problem, but hopefully this unblocks this rule

**Comment by JozefAerts** [comment]
Thanks @nhaydel Nic!
The exception has indeed gone, but e.g. "20121123" is still not recognized as an invalid date. Reason may be that the format without dashes is corresponding to the older version of ISO-8601, but CDISC requires it to contain dashes as separators in the date part. Values like "yesterday" and "2012112a" are catched as being invalid.
Partial dates and datetimes now work perfectly, e.g. "2012", "2012-11", "2012-11-30", "2012-12-13T08", "2013-01-10T09:03" are all accepted. 
I attach the two test files. YAML implementation is already available through the rule editor.
[unit-test-coreid-SENDIG67-negative.xlsx](https://github.com/cdisc-org/cdisc-rules-engine/files/9763795/unit-test-coreid-SENDIG67-negative.xlsx)
[unit-test-coreid-SENDIG67-positive.xlsx](https://github.com/cdisc-org/cdisc-rules-engine/files/9763796/unit-test-coreid-SENDIG67-positive.xlsx)
Can you also please reopen the issue? Alternatively I can open a new one.

**Comment by nhaydel** [comment]
@JozefAerts i would open a new issue since this bug is more nuanced than just iso-8601 validation is not working. This will require some custom logic for requiring dashes in dates. Shouldn’t be too hard, but it is slightly different

**Comment by JozefAerts** [comment]
Thanks Nic, 
I will open a new issue, it is indeed slightly different.
